diff/parse.go

```diff
@@ -341,7 +341,7 @@ func (r *FileDiffReader) ReadExtendedHeaders() ([]string, error) {
 func handleEmpty(fd *FileDiff) (wasEmpty bool) {
 	var err error
 	switch {
-	case (len(fd.Extended) == 3 || len(fd.Extended) == 4 && strings.HasPrefix(fd.Extended[3], "Binary files ")) &&
+	case (len(fd.Extended) == 3 || len(fd.Extended) == 4 && strings.HasPrefix(fd.Extended[3], "Binary files ") || len(fd.Extended) > 4 && strings.HasPrefix(fd.Extended[3], "GIT binary patch")) &&
```

Comment:
looks like this conditional is big enough and repeated enough to be factored out?

Reply:
Made it smaller and replaced the common len check by a variable